packages/experimental-identity-and-auth/src/index.ts

```diff
@@ -0,0 +1,108 @@
+import { HttpRequest, Identity, IdentityProvider } from "@smithy/types";
+
+/**
+ * Interface to get an IdentityProvider for a specified HttpAuthScheme
+ * @internal
+ */
+export interface IdentityProviderConfiguration {
+  /**
+   * Get the IdentityProvider for a specified HttpAuthScheme.
+   * @param schemeId schemeId of the HttpAuthScheme
+   * @returns IdentityProvider or undefined if HttpAuthScheme is not found
+   */
+  getIdentityProvider(schemeId: string): IdentityProvider<Identity> | undefined;
+
+  /**
+   * Gets the configured HttpAuthSchemes.
+   * @returns all configured HttpAuthSchemes
+   */
+  getAuthSchemes(): Map<string, HttpAuthScheme>;
+}
+
+/**
+ * Interface to sign identity and signing properties.
+ * @internal
+ */
+export interface HttpSigner {
+  /**
+   * Signs an HttpRequest with an identity and signing properties.
+   * @param httpRequest request to sign
+   * @param identity identity to sing the request with
+   * @param signingProperties property bag for signing
+   * @returns signed request in a promise
+   */
+  sign(httpRequest: HttpRequest, identity: Identity, signingProperties: Record<string, any>): Promise<HttpRequest>;
```

Comment:
use `Record<string, unknown>` so that the value can be cast or fully runtime checked at the access point with the developer knowingly checking an arbitrary value.